crypto/asn1/a_gentm.c

```diff
@@ -101,7 +101,7 @@ int asn1_generalizedtime_to_tm(struct tm *tm, const ASN1_GENERALIZEDTIME *d)
     if (a[o] == 'Z')
         o++;
     else if ((a[o] == '+') || (a[o] == '-')) {
-        int offsign = a[o] == '-' ? -1 : 1, offset = 0;
+        int offsign = a[o] == '-' ? 1 : -1, offset = 0;
```

Comment:
This appears to have been a bug. You are supposed to subtract the offset to get to GMT; such that (everything HHMM) 0100+0100 is the same as 0000+0000, or for those on the USA east coast, 1900-0500 is the same as 0000+0000